python/pyiceberg/expressions/base.py

```diff
@@ -431,7 +431,13 @@ def _(obj: And, visitor: BooleanExpressionVisitor[T]) -> T:
 @visit.register(In)
 def _(obj: In, visitor: BooleanExpressionVisitor[T]) -> T:
     """Visit an In boolean expression with a concrete BooleanExpressionVisitor"""
-    return visitor.visit_unbound_predicate(predicate=obj)
+    return visitor.visit_unbound_predicate(obj)
+
+
+@visit.register(BoundIn)
+def _(obj: BoundIn, visitor: BooleanExpressionVisitor[T]) -> T:
```

Comment:
I don't think this should be registered for the regular expression visitor, just the bound visitor. The regular visitor should always call `visit_bound_predicate` for all bound predicates.

Reply:
Ah right, makes sense--the whole purpose for the inheritance. Updated!